_posts/cci/2021-08-20-cci-download-stats.md

```diff
@@ -0,0 +1,119 @@
+(not possible to know without registered user data) that this is a single user using a novel access pattern, possibly torrent. This access
+pattern artificially bloats the number of users.
+
+**insert graph showing anomalous access pattern**
+
+Historically, at CEDA, [we have used the pattern of registered users and applied it to unregistered user](need_link_from_graham). 
+As all access is un-registered, we don't have any point of reference in this case.
+
+Wanting to remove these anomalous peaks in user activity, we took to eyeballing the data and came up with a number: 10. It looked that there 
+were very few occasions where the number of users exceeded 10/dataset/download method/day. This "worked" but we wanted to review this and
+generate a threshold through more statistical means.
+
+The access patterns seem to come from China, but this approach should filter out any other similar activity
+
+# Filtering
+
+We thought about different options for filtering; machine learning, statistical analysis, heuristics.
+We chose to investigate using standard deviation using statistical analysis.
+
+This would be calculated by using the formula: *t* = *m* + 1.5 * $\sigma$,
+where t is the threshold, m is the mean, 1.5 is the upper bound and $\sigma$ is the standard deviation. 
+In theory if a value is above this statistical upper threshold then that value can be classified as an anomaly and filtered out.
+
+One challenge in this analysis is the sheer number of log messages to process. ~50 million/year. In order to cope with this and
+develop the approach, we started small and scaled up.
+
+The first attempt at the filter calculated the *t* value per month however this would be misclassify if a month had no spikes in access. 
+This would result in a *t* value around 1, which would dismiss valid data.
+
+Running the analysis over a whole year, 2021, we generated a *t* value of 92.4.
+
+{% include figure.html
+    image_url="assets/img/posts/2021-08-20-cci-download-stats/unfiltered_vs_filtered_hits.png"
+    description="Unfiltered vs Filtered download stats activity"
+%}
+
+With and without the filtering we can see the activity graph is almost normalised to visualise the individual days of activity. 
+Where as before, it was dwarfed by the peaks caused by activities above the t value, 92.4.
+
+The filter operates the same way as the eyeball attempt, cutting off accesses where the number of unique 
+IPs/country/day/download method/dataset is above *t*.
+To visualise the effect of the filtering on an individual dataset level, we grouped the data by country and date with a 
+red line to represent the *t* value. This would clearly show where and when the spikes were occuring on a dataset and country level.
+
+{% include figure.html
+    image_url="assets/img/posts/2021-08-20-cci-download-stats/Single_insight_anamoly_scatter.jpeg"
```

Comment:
- [ ] Add a y-axis title. 
- [ ] Remove the legend
- [ ] Only annotate "x"s above the threshold value